Generated C Code is invalid due to a 'return' statement as function parameter

**V version:** 0.1.10 (latest commit)
**OS:** eOS 5.0 (Linux 5.1.15)

**What did you do?**
Write code that takes in a map and creates a array.
The array has a entry for each key-value pair.

Then I made a function that uses the map to array one and should only join them with `\n`:

```go
pub fn to_string(values map) string {
    return to_string_arr(values).join('\n')
}
```

**What did you expect to see?**
That the result of `to_string_arr()` is joined with `\n` and returned.
 
**What did you see instead?**
```
$ v konftest.v 
/home/atjontv//.vlang//konftest.c:4842:21: error: expected expression
 array_string_join( return  konf__to_string_arr ( values ) , tos2("\n") ) ;
                    ^
1 error generated.
V panic: clang error
```

As one can see, the V compiler placed a `return` in front of the function call for no reason.

----

I managed to fix it by creating a variable holding the result of to_string_arr

```go
pub fn to_string(values map) string {
    array := to_string_arr(values)
    return array.join_lines()
}
```

The source code is here https://github.com/ATVG-Studios/v/commit/6353eff33d1f4432e8b16c882f00cac1d9e72a50

similar problem with this code 

this might give more hints to the root cause 

````
import time

             
fn main() {

    t := time.now()  // type Time 
    println(t.hhmmss())

    st := time.now().hhmmss() // type string    
    println(st)

    println(time.now().hhmmss()) // type string    
}
````
````
/home/work//.vlang//time.c: In function ‘main’:
/home/work//.vlang//time.c:3682:31: error: incompatible type for argument 1 of ‘println’
  time__Time_hhmmss( println ( time__now ( ) ) ) ;
                               ^~~~~~~~~~~~~
/home/work//.vlang//time.c:2270:22: note: expected ‘string’ {aka ‘struct string’} but argument is of type ‘time__Time’ {aka ‘struct time__Time’}
  void println(string s) {
               ~~~~~~~^
V panic: clang error
````

generated C code for testcase : 
````
...
int main(int argc, char** argv) {
 init_consts();
time__Time t= time__now ( ) ;
 println ( time__Time_hhmmss( t ) ) ;
string st= time__Time_hhmmss( time__now ( ) ) ;
 println ( st ) ;

 time__Time_hhmmss( println ( time__now ( ) ) ) ;    // <--------------- !!!!!
// error wrong nested calling sequence  
// sould be  println(time__Time_hhmmss( time__now ( )));

 }
....
 ```

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -1209,6 +1209,7 @@ fn (p mut Parser) name_expr() string {
 	hack_pos := p.scanner.pos
 	hack_tok := p.tok
 	hack_lit := p.lit
+	ph := p.cgen.add_placeholder()
 	// amp
 	 
 	ptr := p.tok == AMP
@@ -1304,7 +1305,7 @@ fn (p mut Parser) name_expr() string {
 			mut typ := p.cast(name)
 			p.gen(')')
 			for p.tok == DOT {
-				typ = p.dot(typ, 0)
+				typ = p.dot(typ, ph)
 			}
 			return typ
 		}
@@ -1405,7 +1406,7 @@ fn (p mut Parser) name_expr() string {
 		mut typ := ''
 		for p.tok == DOT {
 			// println('dot #$dc')
-			typ = p.dot(f.typ, 0)
+			typ = p.dot(f.typ, ph)
 		}
 		return typ
 	}
@@ -1850,7 +1851,7 @@ fn (p mut Parser) expression() string {
 	}
 	if p.tok == DOT  { 
 		for p.tok == DOT {
-			typ = p.dot(typ, 0)
+			typ = p.dot(typ, ph)
 		}
 	}
 	// + - |

